### DISCUSSION TITLE
Sprite.filters.push() "cannot add property 0, object is not extensible"

### DISCUSSION THREAD

**Discussion by steven-twerdochlib** [pixijs/pixijs#10760] [Q&A] [answered]
Hi, 
I'm currently upgrading from pixi v7 to v8 but have come across the bug when running the following line of code:
`currentImageObject.image.filters.push(filter);`
I noticed in the docs filters is a ready-only, is that the reason I can't push to it? See: [https://pixijs.download/release/docs/scene.Sprite.html#filters](url)

**Comment by bigtimebuddy** [comment, marked as answer]
That is correct. You need to assign the array and it cannot be mutated.

**Reply by steven-twerdochlib** [reply]
Thanks for confirming, for those interested this is how I've solved the issue:
```
let tmpFilter = [...currentImageObject.image.filters];
tmpFilter.push(negativeFilter);
currentImageObject.image.filters = tmpFilter;
```

**Reply by bigtimebuddy** [reply]
This also works:
```js
currentImageObject.image.filters = [...currentImageObject.image.filters, negativeFilter];
```
